core/src/main/java/io/confluent/rest/metrics/RestMetricsContext.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2014 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest.metrics;
+
+import io.confluent.rest.RestConfig;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.common.metrics.MetricsContext;
+
+public class RestMetricsContext<T> implements MetricsContext {
+  /**
+   * MetricsContext Label's for use by Confluent's TelemetryReporter
+   */
+  public static final String METRICS_CONTEXT_PREFIX = "metrics.context.";
+  public static final String RESOURCE_LABEL_PREFIX = "resource.";
+  public static final String RESOURCE_LABEL_TYPE = RESOURCE_LABEL_PREFIX + "type";
+
+  /**
+   * Client or Service's metadata map.
+   */
+  private final Map<String, String> metadata;
+
+  /**
+   * {@link io.confluent.rest.Application} {@link MetricsContext} configuration.
+   */
+  public RestMetricsContext(RestConfig config) {
+    /* Copy all configuration properties prefixed into metadata instance. */
+    this(config.originalsWithPrefix(METRICS_CONTEXT_PREFIX));
+
+    /* JMX_PREFIX is synonymous with MetricsContext.NAMESPACE */
+    this.putNamespaceLabel(MetricsContext.NAMESPACE,
+            config.getString(RestConfig.METRICS_JMX_PREFIX_CONFIG));
+
+    /* Never overwrite preexisting resource labels */
+    this.putResourceLabel(RESOURCE_LABEL_TYPE,
+            config.getString(RestConfig.METRICS_JMX_PREFIX_CONFIG));
+  }
+
+  public RestMetricsContext(Map<String, Object> config) {
+    this.metadata = new HashMap<>();
+    config.forEach((key, value) -> metadata.put(key, (String) value));
+  }
+
+  /**
+   * Sets {@link MetricsContext} namespace label.
+   */
+  protected void putNamespaceLabel(String labelKey, String labelValue) {
+    /* Remove resource label if present */
+    this.metadata.put(labelKey.replace(RESOURCE_LABEL_PREFIX, ""),
```

Comment:
Do we want to pass labelKey here? I mean we know namespace key has to be MetricsContext.NAMESPACE ('_namespace').

Reply:
NamespaceKey is probably a bad name. Basically I meant any label bound to the namespace itself in case a downstream component wants to add some additional labels. Perhaps just "Label" would be better.

Reply:
🤦 good catch

Reply:
Went with putLabel, open to suggestions though